third_party/move/move-vm/runtime/src/storage/code_storage.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{loader::Script, logging::expect_no_verification_errors, ModuleStorage};
+use ambassador::delegatable_trait;
+use move_binary_format::{errors::VMResult, file_format::CompiledScript};
+use move_vm_types::{
+    code::{Code, ScriptCache},
+    module_linker_error,
+};
+use sha3::{Digest, Sha3_256};
+use std::sync::Arc;
+
+/// Returns the hash (SHA-3-256) of the bytes. Used for both modules and scripts.
+pub fn compute_code_hash(bytes: &[u8]) -> [u8; 32] {
```

Comment:
Nit: This can be a generic utility function as well

Reply:
Changed the name to `sha3_256`, and moved to `move-vm-types` crate so it serves as a generic utility.